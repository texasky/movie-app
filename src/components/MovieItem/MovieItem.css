.movie-item {
    overflow: hidden;
    transition: transform 0.5s;
}

.movie-item:hover {
    transform: scale(1.2);
}