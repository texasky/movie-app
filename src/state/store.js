import { applyMiddleware, createStore } from "redux"
import configureStore from 'redux-mock-store'
import reducers from "./reducers/index"
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));

export default store; 