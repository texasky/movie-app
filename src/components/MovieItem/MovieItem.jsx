import { useDispatch } from 'react-redux';
import { openMovie } from '../../state/action-creator/index'
import './MovieItem.css'

const MovieItem = (props) => {
    const dispatch = useDispatch();

    const openMovieHandler = () => {
        dispatch(openMovie(props.movie.imdbID))
    }

    return (
        <div className='movie-item' onClick={openMovieHandler}>
            <img src={props.movie.Poster} alt="movie"></img>
        </div>
    )
}

export default MovieItem;