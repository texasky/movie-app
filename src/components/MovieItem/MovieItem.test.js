import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MovieItem from './MovieItem';
import { Provider } from 'react-redux';
import store from './../../state/store'

test('Movie item component tests', () => {
    const clickFn = jest.fn();
    const utils = render(<Provider store={store}><MovieItem onClick={clickFn()}/></Provider>);

    const movieComponent = utils.getByTestId('movie-item');
    fireEvent.click(movieComponent)

    expect(clickFn).toBeCalled();
});

test('Movie item component matches snapshot', () => {
    const clickFn = jest.fn();
    const tree = renderer.create(<Provider store={store}><MovieItem onClick={clickFn()}/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});