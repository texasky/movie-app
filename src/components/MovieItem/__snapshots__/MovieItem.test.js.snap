// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie item component matches snapshot 1`] = `
<div
  className="movie-item"
  data-testid="movie-item"
  onClick={[Function]}
>
  <img
    alt="movie"
  />
  <div
    className="movie-item-title"
  />
</div>
`;
